jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -167,75 +169,133 @@ protected Callback getCallback()
         }
 
         protected void process()
+        {
+            process(getRequest(), getResponse(), getCallback());
+        }
+
+        protected boolean process(Request request, Response response, Callback callback)
         {
             try
             {
-                if (!getHandler().handle(getRequest(), getResponse(), getCallback()))
-                    Response.writeError(getRequest(), getResponse(), getCallback(), HttpStatus.NOT_FOUND_404);
+                if (getHandler().handle(request, response, callback))
+                    return true;
+
+                Response.writeError(getRequest(), getResponse(), getCallback(), HttpStatus.NOT_FOUND_404);
```

Comment:
Shouldn't that be
```java
Response.writeError(request, response, callback, HttpStatus.NOT_FOUND_404);
```
instead?

Reply:
If the handle with the potentially wrapped request/response/callback fail, then I want to write the error with the unwrapped versions.  I've added comments.